tests/k8st/tests/test_bgp_advert.py

```diff
@@ -273,3 +325,68 @@ def check_routes_gone():
                 for cip in cluster_ips:
                     self.assertNotIn(cip, routes)
             retry_until_success(check_routes_gone, retries=10, wait_time=5)
+
+    def test_rr_filtering_svc_adv(self):
+        """
+        Test route with svc ip is not filtered on RR
+        """
+        with DiagsCollector():
+            # Create a network policy on node-A accepts traffic from the external node.
+            run("""docker exec -i kube-master kubectl apply -f - << EOF
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: allow-tcp-80-ex
+  namespace: bgp-test
+spec:
+  podSelector: {node-A}
+  policyTypes:
+  - Ingress
+  ingress:
+  - from:
+    - ipBlock: { cidr: 20.192.0.5/32 }
+    ports:
+    - protocol: TCP
+      port: 80
+EOF
+""")
+           # Create RR host on node-B
+            cluster_id = str(IPAddress(0xFF000001))
+            rr = DockerHost('RR', start_calico=False)
+            rr.add_resource({
+                'apiVersion': 'projectcalico.org/v3',
+                'kind': 'Node',
+                'metadata': {
+                    'name': rr.get_hostname(),
+                    'labels': {
+                        'routeReflectorClusterID': cluster_id,
+                        },
+                    },
+                'spec': {
+                    'bgp': {
+                        'routeReflectorClusterID': cluster_id,
+                        },
+                    },
+                })
+            rr.start_calico_node()
```

Comment:
(As I've written elsewhere:) using DockerHost and rr.add_resource is wrong here because the compute nodes already exist in the k8st setup. So instead you need to get the Node resource of one of the existing nodes and modify it so that that node behaves as a RR. I experimented with code for some of that at https://github.com/projectcalico/node/pull/151, specifically

```
kubectl exec -i -n kube-system calicoctl -- /calicoctl get node kube-node-2 -o json
```

and you can see an example of what that does in the output for Job #2 at https://semaphoreci.com/calico/node/branches/pull-request-151/builds/3.

Building on that, you will still need to:
- get the JSON for the node into a Python dict form, instead of just printing it out into the log
- make the changes (to the dict) needed to configure RR behaviour
- apply the modified JSON back into Calico (with `calicoctl apply`)